addons/website_slides/models/res_partner.py

```diff
@@ -10,9 +10,23 @@ class ResPartner(models.Model):
     slide_channel_ids = fields.Many2many(
         'slide.channel', 'slide_channel_partner', 'partner_id', 'channel_id',
         string='eLearning Courses')
+    slide_channel_completed_ids = fields.One2many(
+        'slide.channel', string='Completed Courses', compute='_compute_slide_channel_completed_ids',
+        search='_search_slide_channel_completed_ids')
     slide_channel_count = fields.Integer('Course Count', compute='_compute_slide_channel_count')
     slide_channel_company_count = fields.Integer('Company Course Count', compute='_compute_slide_channel_company_count')
 
+    def _compute_slide_channel_completed_ids(self):
+        for partner in self:
+            partner.slide_channel_completed_ids = self.env['slide.channel.partner'].search([
+                ('partner_id', '=', partner.id),
+                ('completed', '=', True)
+            ]).mapped('channel_id')
+
+    def _search_slide_channel_completed_ids(self, operator, value):
+        cp_done = self.env['slide.channel.partner'].search([('channel_id', operator, value), ('completed', '=', True)])
```

Comment:
@pko-odoo What about the search_read suggestion? Did you try it?

Side note: Runbot is red, might want to check why.

Reply:
When using the search_read method I get a list of dictionary such as `[{'id': 9, 'partner_id': (8, 'YourCompany, Joel Willis')}]` so I'm not sure it's really helpful since the search_read method does a search in the first place.
Or maybe I use it the wrong way ?

Reply:
It should be the same as reading on ['channel_id'] here above, and it seems to work there?
If it makes the code more complicated, just leave it as is.

Reply:
Actually the code above is not working.. As you said it needs a recordset
I reverted that part in the last commit